packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -0,0 +1,858 @@
+    session: Session,
+    redis: Redis,
+    tx_info: GetTransactionResp,
+    tx_sig,
+    timestamp,
+    challenge_event_bus: ChallengeEventBus,
+):
+    logger.debug(f"index_payment_router.py | Processing tx={tx_info.to_json()}")
+    result = tx_info.value
+    if not result:
+        logger.error("index_payment_router.py | No result")
+        return
+    meta = result.transaction.meta
+    if not meta:
+        logger.error("index_payment_router.py | No result meta")
+        return
+    error = meta.err
+    if error:
+        logger.error(
+            f"index_payment_router.py | Skipping error transaction from chain {tx_info}"
+        )
+        return
+    transaction = result.transaction.transaction
+    if not hasattr(transaction, "message"):
+        logger.error(
+            f"index_payment_router.py | No transaction message found {transaction}"
+        )
+        return
+
+    tx_message = cast(Message, transaction.message)
+    account_keys = list(map(lambda x: str(x), transaction.message.account_keys))
+
+    # Check for valid instruction
+    has_route_instruction = has_log(meta, "Program log: Instruction: Route")
+
+    instruction = get_valid_instruction(tx_message, meta, PAYMENT_ROUTER_ADDRESS)
+    if instruction is None:
+        logger.error(f"index_payment_router.py | {tx_sig} No Valid instruction found")
+        return
+
+    # TODO: Parse existing TransferChecked instruction first to get the address which sent
+    # money _into_ the payment router. This will be necessary to correctly index
+    # external transfers via Payment Router from a userbank.
+
+    if has_route_instruction:
+        process_route_instruction(
+            session=session,
+            redis=redis,
+            instruction=instruction,
+            memos=decode_all_solana_memos(tx_message=tx_message),
+            account_keys=account_keys,
+            meta=meta,
+            tx_sig=tx_sig,
+            slot=result.slot,
+            challenge_event_bus=challenge_event_bus,
+            timestamp=timestamp,
+        )
+
+
+def process_payment_router_txs() -> None:
+    solana_client_manager: SolanaClientManager = (
+        index_payment_router.solana_client_manager
+    )
+    challenge_bus: ChallengeEventBus = index_payment_router.challenge_event_bus
+    db = index_payment_router.db
+    redis = index_payment_router.redis
+    logger.info("index_payment_router.py | Acquired lock")
+
+    # Exit if required configs are not found
+    if not check_config():
+        return
+
+    # List of signatures that will be populated as we traverse recent operations
+    transaction_signatures = []
+
+    last_tx_signature = None
+    latest_global_slot = None
+
+    # Loop exit condition
+    intersection_found = False
+    is_initial_fetch = True
+
+    # Get the latests slot available globally before fetching txs to keep track of indexing progress
+    try:
+        latest_global_slot = solana_client_manager.get_slot()
+    except:
```

Comment:
nit, grab the exception and log it probably

Reply:
if this fails should we abort? while not technically required, why continue? something's probably wrong

Reply:
That's a fair point. I would expect subsequent solana calls to also not be reliable.